ssh/tailssh/tailssh.go

```diff
@@ -87,7 +87,7 @@ func init() {
 
 // HandleSSHConn handles a Tailscale SSH connection from c.
 func (srv *server) HandleSSHConn(c net.Conn) error {
-	ss, err := srv.newSSHServer()
+	ss, err := srv.newSSHServer(c)
```

Comment:
we pass this now earlier, as we'll use its identity to conditionally configure the ssh.Server